lib/services/content_store.rb

```diff
@@ -0,0 +1,43 @@
+class Services::ContentStore
+  def hard_refresh_cache
+    finder_paths.map do |base_path|
+      item = content_item(base_path)
+      Rails.cache.write(cache_key(base_path), item)
+    end
+  end
+
+  def soft_refresh_cache
+    finder_paths.map do |base_path|
+      cached_content_item(base_path)
+    end
+  end
+
+  def cached_content_item(base_path)
+    Rails.cache.fetch(cache_key(base_path)) do
+      content_item(base_path)
+    end
+  end
+
+  def cache_key(base_path)
+    "finder-frontend_content_items#{base_path}"
+  end
+
+private
+
+  def content_item(base_path)
+    GovukStatsd.time("content_store.fetch_request_time") do
+      content_store.content_item(base_path).to_h
+    end
+  rescue GdsApi::HTTPServerError, GdsApi::HTTPBadGateway
```

Comment:
You probably want this to be `Gds::HTTPIntermittentServerError`. In the case above `GdsApi::HTTPBadGateway` has no effect as it is a subclass of `GdsApi::HTTPSserverError`.